cmd/provenanced/config/interceptor.go

```diff
@@ -58,6 +58,9 @@ func InterceptConfigsPreRunHandler(cmd *cobra.Command) error {
 	//  2. The config is packed and we're filling in the missing with defaults.
 	if vpr.GetBool(EnvTypeFlag) {
 		DefaultKeyringBackend = "test"
+	} else {
+		// Hard-code the consensus.timeout_commit value for non-testnets.
+		vpr.Set("consensus.timeout_commit", "3.5s")
```

Comment:
The "3.5s" isn't a magic value, it's just what we want the setting to be. And nothing else cares about that value, so it wasn't yet worth putting in a variable.

Making note of this in the comment is overkill. There's tons of stuff that happens in `InterceptConfigsPreRunHandler` that isn't specifically called out in the comment, and there's too much to put in there anyway.

The pre-upgrade command's primary purpose is to update the config files to the new formats. If we want to change config values, we can do it there too, but there's no guarantee that a node operator runs that command ever. So we can't just use that to set everyone's config value to what we want it. At one point, we did use the pre-upgrade command to change the values to 1.5s, but not every node got that, and the varying values are likely causing us problems, which is why it's now being hard-coded. This does not conflict with that at all.

The default value is staying unchanged so that it's what gets used for testnets (absent any other config). We don't want to change that, only the value that gets used on mainnet nodes.

The config warning system will not get triggered by this since the upper limit is the default (1.5s) plus 2s = 3.5s. The warning is only issued if the configured value is strictly greater than that upper limit, which 3.5s is not. But you do bring up a good point, that we should be giving a warning if it's not 3.5s. I'll push up a change in a bit.

Unfortunately, the place to document this hard-coding is in the config file template. But that template is defined in a library and we can't change it. This config field should be consistent on all nodes and probably shouldn't have been part a node's config anyway. No one is changing it other than at our suggestion, so there probably won't be much confusion.